sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConnectionStringProperties.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.implementation.ConnectionStringProperties;
+import com.azure.core.credential.TokenCredential;
+
+import java.net.URI;
+
+/**
+ * A utility class that parses a connection string into sections. A Event Hub connection string is a set of key value
+ * pairs separated by semi-colon. A typical example is
+ * {@code "Endpoint=sb://foo.EventHub.windows.net/;SharedAccessKeyName=someKeyName;SharedAccessKey=someKeyValue"}.
+ *
+ * <p>
+ * A connection may have the following sections:
+ * <ul>
+ *     <li>Endpoint, which is mandatory. The hostname part of it is the "Fully qualified namespace".</li>
+ *     <li>SharedAccessKeyName and SharedAccessKey, optional, used to authenticate the access to the EventHub.</li>
+ *     <li>SharedAccessSignature, optional, an alternative way to authenticate the access to the EventHub.</li>
+ *     <li>EntityPath, optional, the queue name or the topic name under the service namespace</li>
+ * </ul>
+ *
+ * <p>
+ * When you have a EventHub connection string, you can use {@link EventHubClientBuilder#connectionString(String)}
+ * to build a client. If you'd like to use a {@link TokenCredential} to access a Event Hub, you can use this utility
+ * class to take the fully qualified namespace and optionally the entity path (queue/topic name) from the connection
```

Comment:
EntityPath is not optional in the case with a TokenCredential.

```suggestion
 * class to get the fully qualified namespace and entity path from the connection
```

Reply:
According to your comment, fixed in new version.